File: src/dataclass/Project.cpp
// Project.cpp

#include <string>
#include <sstream>
#include <vector>
#include <map>

#include "dataclass/Project.hpp"

std::string ft::dataclass::Project::get_str(){
    std::ostringstream oss;

    oss << "--- Song Information ---\n";
    oss << "title = " << title << "\n";
    oss << "author = " << author << "\n";
    oss << "copyright = " << copyright << "\n";
    
    oss << "--- Comment ---\n";
    oss << comment << "\n";
    
    oss << "--- Global Settings ---\n";
    oss << "machine = " << machine << "\n";
    oss << "framerate = " << framerate << "\n";
    oss << "expansion = " << expansion << "\n";
    oss << "vibrato = " << vibrato << "\n";
    oss << "split = " << split << "\n";
    oss << "n163channels = " << n163channels << "\n"; 

    // oss << "--- Macros ---\n";
    // oss << "--- Grooves ---\n";
    // oss << "--- DPCM ---\n";
    // oss << "--- KeyDPCM ---\n";
    // oss << "--- Instruments ---\n";
    // oss << "--- Tracks ---\n";
    
    return oss.str();
}

File: src/dataclasses/project.py
# project.py

class Project:
    def __init__(self):
        self.title = ""
        self.author = ""
        self.copyright = ""
        self.comment = ""
        self.machine = 0
        self.framerate = 0 
        self.expansion = 0 
        self.vibrato = 1
        self.split = 32
        self.n163channels = 0
        self.macros = {}
        self.samples = {}
        self.grooves = {}
        self.instruments = {}
        self.tracks = {}

    def __str__(self) -> str:
        return "{}".format(self.__dict__)
    
    def __repr__(self) -> str:
        return self.__str__()


